include/rx/ranges.hpp

```diff
@@ -1655,6 +1656,107 @@ struct reverse {
     }
 };
 
+/// Sliding window over an input range.
+///
+/// E.g. for a range `(1,2,3,4,5)` a window of 2 elements is `[(1,2), (2,3), (3,4), (5)]`.
```

Comment:
do you mean `[(1,2), (2,3), (3,4), (4,5)]` ?

Reply:
Actually, no. Otherwise the output would be the same for an input `(1,2,3,4,5)` and `(1,2,3,4,4,5)`.

I was not sure how to handle "underfull" groups, whether to use empty optional<>s, a padding element, or let the user decide. Returning a range with too fewer elements is the simplest solution, so I chose to use that. Maybe the other options could be added later.

Reply:
with `(1,2,3,4,4,5)` I would expect `(1,2),(2,3),(3,4),(4,4),(4,5)` 🤔.

Reply:
Argh. I'm dumb, and I've confused myself. :) Sorry, I'll fix the description.